Dockerfile.rpm

```diff
@@ -10,13 +10,15 @@ ARG REDHAT_USERNAME
 ARG REDHAT_PASSWORD
 ARG RHEL="false"
 
-RUN if [ "$RHEL" = "true" ] ; then subscription-manager register --username ${REDHAT_USERNAME} --password ${REDHAT_PASSWORD} --auto-attach ; fi
-RUN if [ "$RHEL" = "true" ] ; then yum-config-manager --enable 'rhel-*-server-optional-rpms' ; fi
+RUN if [ "$RHEL" = "true" ] ; then subscription-manager register --username ${REDHAT_USERNAME} --password ${REDHAT_PASSWORD} ; fi
+RUN if [ "$RHEL" = "true" ] ; then subscription-manager refresh ; fi
+RUN if [ "$RHEL" = "true" ] ; then subscription-manager attach --auto ; fi
+RUN if [ "$RHEL" = "true" ] ; then command -v yum-config-manager && yum-config-manager --enable 'rhel-*-server-optional-rpms' || true ; fi
 
-RUN yum -y install wget tar readline-devel pcre-devel openssl-devel gcc curl unzip \
-  zlib-devel make gcc gcc-c++ bzip2 patch perl m4 git coreutils which
+RUN yum -y --skip-broken install wget tar readline-devel pcre-devel openssl-devel gcc curl unzip \
+  zlib-devel make gcc gcc-c++ bzip2 patch perl m4 git coreutils
 
-RUN which realpath || /bin/bash -c "curl -sSLO https://ftp.tu-chemnitz.de/pub/linux/dag/redhat/el6/en/x86_64/rpmforge/RPMS/realpath-1.17-1.el6.rf.x86_64.rpm && rpm -i realpath-1.17-1.el6.rf.x86_64.rpm"
+RUN command -v realpath || /bin/bash -c "curl -sSLO https://ftp.tu-chemnitz.de/pub/linux/dag/redhat/el6/en/x86_64/rpmforge/RPMS/realpath-1.17-1.el6.rf.x86_64.rpm && rpm -i realpath-1.17-1.el6.rf.x86_64.rpm"
```

Comment:
Stack Overflow pointed out that `command -v` is readily available everywhere while `which` is not